exercises/luhn/description.md

```diff
@@ -1,53 +1,112 @@
-  3554" in response to "2323 2005 7766 355".
-
-## About Checksums
-
-A checksum has to do with error-detection. There are a number of different
-ways in which a checksum could be calculated.
-
-When transmitting data, you might also send along a checksum that says how
-many bytes of data are being sent. That means that when the data arrives on
-the other side, you can count the bytes and compare it to the checksum. If
-these are different, then the data has been garbled in transmission.
-
-In the Luhn problem the final digit acts as a sanity check for all the prior
-digits. Running those prior digits through a particular algorithm should give
-you that final digit.
-
-It doesn't actually tell you if it's a real credit card number, only that it's
-a plausible one. It's the same thing with the bytes that get transmitted --
-you could have the right number of bytes and still have a garbled message. So
-checksums are a simple sanity-check, not a real in-depth verification of the
-authenticity of some data. It's often a cheap first pass, and can be used to
-quickly discard obviously invalid things.
+The [Luhn algorithm](https://en.wikipedia.org/wiki/Luhn_algorithm) is
+a simple checksum formula used to validate a variety of identification
+numbers, such as credit card numbers and Canadian Social Insurance
+Numbers.
 
+Validating a Number
+------
+
+As an example, here is a valid (but fictitious) Canadian Social Insurance
+Number.
+
+```
+046 454 286
+```
+
+The first step of the Luhn algorithm is to double every second digit,
+starting from the right. We will be doubling
+
+```
+_4_ 4_4 _8_
+```
+
+If doubling the number results in a number greater than 9 then subtract 9
+from the product. The results of our doubling:
+
+```
+086 858 276
+```
+
+Then sum all of the digits
+
+```
+0+8+6+8+5+8+2+7+6 = 50
+```
+
+If the sum is evenly divisible by 10, then the number is valid. This number is valid!
+
+An example of an invalid Canadian SIN where we've changed the final digit
+
+```
+046 454 287
+```
+
+Double the second digits, starting from the right
+
+```
+086 858 276
+```
+
+Sum the digits
+
+```
+0+8+6+8+5+8+2+7+7 = 51
+```
+
+51 is not evenly divisible by 10, so this number is not valid.
+
+----
+
+An example of an invalid credit card account
+
+```
+8273 1232 7352 0569
+```
+
+Double the second digits, starting from the right
+
+```
+7253 2262 5312 0539
+```
+
+Sum the digits
+
+```
+7+2+5+3+2+2+5+2+5+3+1+2+0+5+3+9 = 57
+```
+
+57 is not evenly divisible by 10, so this number is not valid.
+
+Generating a Check Digit
+----
+
+The final number in these examples is called to a [check digit](https://en.wikipedia.org/wiki/Check_digit).
+
+```
+ 046 454 28   6
+| account |   |check digit|
+```
+
+```
+8273 1232 7352 056   9
+|    account     |   |check digit|
+```
+
+Given an account number your code should be able to return a check digit
+that can be appended to the account to generate a valid Luhn
```

Comment:
> a valid Luhn

You will have to excuse a question from someone unfamiliar with the domain, but is it valid to use here "Luhn" as a shorthand for "number that is valid according to Luhn algorithm"?